### PR TITLE
Add featured question section to shadow question page

### PR DESCRIPTION
**Pull request by CleverFool77**
Fixes https://github.com/publiclab/plots2/issues/5733 part

Screenshot from 2019-06-08 22-57-13

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by CleverFool77** [comment]
Hi @jywarren @gautamig54 Just added the featured question section.
Thanks !!

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -27,6 +27,15 @@ def index
       .paginate(page: params[:page], per_page: 24)
   end
 
+  def index_shadow
+    @title = 'Questions and Answers'
+    set_sidebar
```
Comment (jywarren) [inline app/controllers/questions_controller.rb]: Is this line maybe not needed?

Comment (CleverFool77) [reply app/controllers/questions_controller.rb]: Yes. I'll fix it.

**Comment by CleverFool77** [comment]
For now, I have followed the same method for shadow pages as earlier controllers for real pages.

**Comment by CleverFool77** [comment]
I tweaked frontend a bit and different cells being called from `/questions` and `/dashboard` were changed as per the info needed.

**Comment by gautamig54** [comment]
Great work @CleverFool77 !!
It will be good if you can decrease the font size of the question in the cards as it will look absurd in case of long questions. 
Also, I think we need to add ellipsis here as well to avoid overflow of questions.

File: app/controllers/questions_controller.rb (hunk 1 above)
Comment (CleverFool77) [reply app/controllers/questions_controller.rb]: Hi @jywarren Did you mean we don't need index_shadow in controller ?
Thanks !!

**Comment by CleverFool77** [comment]
> Great work @CleverFool77 !!
> It will be good if you can decrease the font size of the question in the cards as it will look absurd in case of long questions.
> The heading and the cards are not aligned, can you do that as well.
> Also, I think we need to add ellipsis here as well to avoid overflow of questions.

Hi @gautamig54 I guess in design The question went to second line in card.

**Comment by CleverFool77** [comment]
Screenshot from 2019-06-10 17-09-09

File: app/controllers/questions_controller.rb (hunk 1 above)
Comment (CleverFool77) [reply app/controllers/questions_controller.rb]: But here that would be the method doing whole stuff in this PR.

**Comment by CleverFool77** [comment]
Hi @jywarren I just pushed the latest changes.
Thanks !!

File: app/controllers/questions_controller.rb (hunk 1 above)
Comment (jywarren) [reply app/controllers/questions_controller.rb]: just the `set_sidebar` part, actually, sorry for ambiguity!

**Comment by jywarren** [comment]
Looking good, added clarification to my inline question... @gautamig54 can you approve this PR too and both ping me (or another maintainer) when it's ready? Great work!!!

And, just for quick reference, can you point me at the URL where we'll be able to test this on stable when the time comes? Thank you!!!

**Comment by CleverFool77** [comment]
hi @jywarren @gautamig54  Just pushed the latest changes.
regarding the test, did you mean the main url `/question_shadow` ?

**Comment by gautamig54** [comment]
@CleverFool77 Can add a screenshot with testing question to be greater that two lines? Say 15 words?

**Comment by CleverFool77** [comment]
Hi @gautamig54
Its there in flask notice while Filling up the form for posting qiestion that when title is exceeded more than 8-9 words. It gives message that `way too long titile`.

**Comment by CleverFool77** [comment]
Hi @gautamig54 what do you say about my thought above? 
Maybe we can forward the progress of this PR then.
Hi @jywarren 
Can we have this PR merged first. 
Regarding question size,
Do we need overflow fixing for the question size in frontend? Or a restriction in backend? 
If yes, I'll fix this in next PR.
If no, then I guess it's usually understood for question to be not too long? 
And importantly how many words does the questions usually consist of?
As If fix is needed, then I might work on size of question based on words.

**Comment by CleverFool77** [comment]
Seems like good idea sagarpreet. So maybe truncating the title of question to two-three lines with ellipsis in end?

**Comment by sagarpreet-chadha** [comment]
Thank you and yes sounds great!

**Comment by gautamig54** [comment]
Restricting question length to just 8-9 words may be less. We should restrict it to atleast 20 words? I don't think 20 words is like a too long for a question. 
Also, adding ellipsis(...) for longer question is the best suitable solution.

**Comment by CleverFool77** [comment]
Sure gautami. Lets have this merged first. Till then I'm looking over the matter of overflow.
Thanks !!

**Comment by CleverFool77** [comment]
The screenshot feature looks super cool :open_mouth: :open_mouth: . Damnnnn :heart_eyes: !!!

**Comment by jywarren** [comment]
Do you wanna add the questions page? https://github.com/publiclab/plots2/tree/master/test/system/screenshots_test.rb

**Comment by CleverFool77** [comment]
Sure I'll add this for shadow route for now.

**Comment by CleverFool77** [comment]
Hi @jywarren just pushed the latest changes with the addition of test screenshot for question and shadow route.
Thanks.

**Comment by jywarren** [comment]
Oooooh very cool!!!

File: app/views/questions/_questions_shadow.html.erb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      <% end %>
+
+      <div class="col-md-2">
+      </div>
+  </div>
+</div>
+
+<style>
+
+.card {
```
Comment (jywarren) [inline app/views/questions/_questions_shadow.html.erb]: Should we start to consolidate these styles and re-use them across cards? What do you think?

Comment (CleverFool77) [reply app/views/questions/_questions_shadow.html.erb]: Though the cards looks same but still there are many differences. The positioning and size of cards in different pages varies.

**Comment by gautamig54** [comment]
This looks great @CleverFool77! Let's get this one merged and move ahed with the recent questions section.

File: app/views/questions/_questions_shadow.html.erb (hunk 2 above)
Comment (jywarren) [reply app/views/questions/_questions_shadow.html.erb]: OK, I still am thinking perhaps we should make a planning item for moving all the re-usable card layouts to one template folder so they're all next to one another. They are hard to find! Like `/app/views/items` or `/app/views/cards` maybe?

Comment (jywarren) [reply app/views/questions/_questions_shadow.html.erb]: But doesn't have to be in this PR

Comment (CleverFool77) [reply app/views/questions/_questions_shadow.html.erb]: Sure. Should I do this in next PR ?

Comment (CleverFool77) [reply app/views/questions/_questions_shadow.html.erb]: Or Le'ts wait a bit for all the cards and then shift them altogether to `/app/views/cards`

**Comment by jywarren** [comment]
OK, either option for /views/cards is fine, just want us to be thinking
ahead. Thanks!!!

**Pull request merged**